app/models/saved_claim/veteran_readiness_employment_claim.rb

```diff
@@ -218,6 +219,50 @@ def send_to_res(user)
     service.submit
   end
 
+  def add_errors_from_form_validation(form_errors)
+    form_errors.each do |e|
+      errors.add(e[:fragment], e[:message])
+      e[:errors]&.flatten(2)&.each { |nested| errors.add(nested[:fragment], nested[:message]) if nested.is_a? Hash }
+    end
+    unless form_errors.empty?
+      Rails.logger.error('SavedClaim form did not pass validation',
+                         { form_id:, guid:, errors: form_errors })
+    end
+  end
+
+  def log_schema_errors(schema_errors)
+    unless schema_errors.empty?
+      Rails.logger.error('SavedClaim schema failed validation! Attempting to clear cache.',
+                         { form_id:, errors: schema_errors })
+      true
+    end
+  end
+
+  def form_matches_schema
+    return unless form_is_string
+
+    schema = VetsJsonSchema::SCHEMAS[self.class::FORM]
+    schema_v2 = VetsJsonSchema::SCHEMAS[self.class::FORMV2]
+
+    schema_errors = validate_schema(schema)
+    clear_cache = log_schema_errors(schema_errors)
+
+    schema_v2_errors = validate_schema(schema_v2)
+    clear_cache_v2 = log_schema_errors(schema_v2_errors)
+
+    validation_errors = validate_form(schema, clear_cache)
```

Comment:
Before I pulled master, I saw the method defined as:
```
  def validate_form(schema, _clear_cache)
    errors = JSONSchemer.schema(schema).validate(parsed_form).to_a
    return [] if errors.empty?

    reformatted_schemer_errors(errors)
  end
```
Which I thought was odd. Why have the `_clear_cache` param if it does nothing? But then I pulled and it looks like Brandon Cooper changed it an hour ago, so I think you may have some merge conflicts. Seems like you can just remove the `clear_cache` param. But... doesn't that also mean we're not really clearing the cache? We're just logging the fact that we're clearing the cache if there are errors, but I don't see any cache being cleared.